packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -566,7 +566,18 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
       } catch (err) {
         const isAlreadyKnown = /AlreadyKnown/gi.test(err.message)
         const isFeeTooLow = /FeeTooLowToCompete/gi.test(err.message)
+        const nonceTooLow = /(nonce.*too low|already been used|NONCE_EXPIRED)/gi.test(err.message)
         const shouldRetry = (isAlreadyKnown || isFeeTooLow) && i < maxRetries
+        if (nonceTooLow) {
+          if (!this.nonce) {
```

Comment:
My understanding is that there is no `this.nonce` for the first tx, but if it is boosted then it will exist because it will be passed into the constrcutor.

If that is correct, I think this stops us handling the issue for non-boosted transactions. I would imagine we want to handle those. Do I have a bad assumption somewhere?

Reply:
Correct, if it is boosted then the nonce would be set and not reach this part of code. The thinking was that nonce should not be overridden if it has been set because it's possible that another transaction could already have the same nonce and we don't want to override another transaction or have the current transaction be replaced by the other transaction. Do you think the nonce check if statement here should be removed?